### PR TITLE
Added: Azure SSO and Captcha

### PR DESCRIPTION
**Pull request by joshuasir**
# Description
This pull request adds Azure SSO (Single Sign-On) login functionality and implements reCAPTCHA v2 hidden to enhance security in the Snipe-IT asset management system. These additions provide easier authentication for organizations using Azure Active Directory and add an extra layer of protection against automated attacks.

## Azure SSO Login
The Azure SSO implementation allows users to log in to Snipe-IT using their Azure AD credentials. This feature streamlines the authentication process and improves security by leveraging Azure's identity management.

## reCAPTCHA v2 Hidden
The addition of reCAPTCHA v2 hidden provides protection against bots and automated attacks without requiring user interaction. This invisible CAPTCHA is attached to reuglar log in button and only prompts users for verification if suspicious activity is detected.

image

Added #15625 (Please replace with the actual issue number if applicable)

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?
- [x] Test A: Azure SSO login flow tested with multiple Azure AD accounts
- [x] Test B: reCAPTCHA v2 hidden tested on login and registration forms
- [x] Test C: Integration tests for Azure SSO authentication
- [x] Test D: Security tests to ensure reCAPTCHA v2 hidden is functioning correctly

**Test Configuration**:
* PHP version: 8.1.6
* MySQL version: 8.1.6
* Webserver version: Apache 2.4.53
* OS version: Windows 11

# Checklist:
- [x] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [x] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by snipe** [comment]
Hi there - thanks for this! Wouldn't the expectation be to let Azure's UI handle the recaptcha side of things? (Also can you please re-target to master, per the developer docs?)

File: resources/views/auth/login.blade.php [hunk 1]
```diff
@@ -94,7 +94,16 @@
...
+                                <button class="g-recaptcha btn btn-primary btn-block" 
+                                data-sitekey="{{ config('recaptcha.site_key') }}"
+                                data-callback='onSubmit'
+                                data-action='submit'>{{ trans('auth/general.login')  }}</button>
+                                <a href="{{ route('login.azure') }}" class="btn btn-primary" style="margin-top: 3px; width: 100%; background-color: white;color: black; outline: gray; border: solid 1px #c8c3c3; display:flex; justify-content:center; align-items:center">
+                                    <svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" width="25" height="25" viewBox="0 0 48 48" style="margin-inline:2px">
+                                        <path fill="#ff5722" d="M6 6H22V22H6z" transform="rotate(-180 14 14)"></path><path fill="#4caf50" d="M26 6H42V22H26z" transform="rotate(-180 34 14)"></path><path fill="#ffc107" d="M26 26H42V42H26z" transform="rotate(-180 34 34)"></path><path fill="#03a9f4" d="M6 26H22V42H6z" transform="rotate(-180 14 34)"></path>
+                                        </svg>Login with Microsoft
```
Comment (snipe) [inline resources/views/auth/login.blade.php:105]: Can we get this into a translation string please?

File: resources/views/auth/login.blade.php [hunk 2]
```diff
@@ -118,4 +127,13 @@
         </div> <!-- end container -->
     </form>
 
-@stop
\ No newline at end of file
+@stop
+
+@push('js')
+<script src="https://www.google.com/recaptcha/api.js"></script>
```
Comment (snipe) [inline resources/views/auth/login.blade.php:133]: Some folks run Snipe-IT in a way that it cannot access the outside world, so this script would not load. (Also, we try not to require Google products, or any other type of commercial 3rd party libraries, although we do allow for google maps.) We'd at least want to check for a recaptcha key before trying to load this.

**Comment by joshuasir** [comment]
yes, correct! I only added recaptcha for the non SSO log in. Sure i can
redirect it.

**Comment by snipe** [comment]
I've been boned by the silent recaptcha before, since I run over a VPN from Portugal - additionally, we already have throttling and BFP on the login/forgotten password pages.

**Comment by snipe** [comment]
Ack, sorry, I meant re-target to develop. Not enough coffee this morning.

**Comment by snipe** [comment]
> yes, correct! I only added recaptcha for the non SSO log in

Okay, great. I do worry that the "magic" that powers it often gets it wrong (as I mentioned), but I guess we can leave that up to folks to configure. Just not sure it's especially needed since we already have throttling.

**Comment by joshuasir** [comment]
> I've been boned by the silent recaptcha before, since I run over a VPN from Portugal - additionally, we already have throttling and BFP on the login/forgotten password pages.

I see your point here, but I believe reCAPTCHA can still provide a benefit here, especially for more sophisticated attacks such as distributed ones or credential stuffing.

**Comment by joshuasir** [comment]
I've opened a new pull request for the dev branch. If you have time, I'd appreciate your review, thank you.

**Pull request closed without merging**